service/controller/resource/promxy/resource.go

```diff
@@ -0,0 +1,152 @@
+package promxy
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/giantswarm/k8sclient/v4/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/discovery/kubernetes"
+	"github.com/prometheus/prometheus/pkg/relabel"
+	"gopkg.in/yaml.v2"
```

Comment:
Please instead use https://github.com/ghodss/yaml

It's a drop in replacement and handles better empty fields, we use this everywhere